financial-services-accelerator/components/org.wso2.financial.services.accelerator.gateway/src/main/java/org/wso2/financial/services/accelerator/gateway/executor/impl/consent/ConsentEnforcementExecutor.java

```diff
@@ -0,0 +1,318 @@
+    private static final String ERROR_MESSAGE = "errorMessage";
+    private static final String HTTP_CODE = "httpCode";
+    private static final String MODIFIED_PAYLOAD = "modifiedPayload";
+    private static final String CONSENT_INFO = "consentInformation";
+    private static volatile String consentValidationEndpoint;
+    private static volatile Key key;
+
+    /**
+     * Method to handle request.
+     *
+     * @param fsApiRequestContext FS request context object
+     */
+    @Generated(message = "Unit testable components are covered")
+    @Override
+    public void preProcessRequest(FSAPIRequestContext fsApiRequestContext) {
+
+    }
+
+    /**
+     * Method to handle post request.
+     *
+     * @param fsApiRequestContext FS request context object
+     */
+    @Override
+    public void postProcessRequest(FSAPIRequestContext fsApiRequestContext) {
+        // Consent ID is required for consent enforcement. If the consent ID is null, we are assume this is a
+        // pre-consent creation call. Therefore consent enforcement is not required.
+        if (fsApiRequestContext.isError() || fsApiRequestContext.getConsentId() == null) {
+            return;
+        }
+
+        Map<String, String> requestHeaders = fsApiRequestContext.getMsgInfo().getHeaders();
+        Map<String, Object> additionalParams = new HashMap<>();
+        additionalParams.put(ELECTED_RESOURCE_TAG, fsApiRequestContext.getMsgInfo().getElectedResource());
+        additionalParams.put(CONSENT_ID_TAG, fsApiRequestContext.getConsentId());
+        additionalParams.put(USER_ID_TAG, fsApiRequestContext.getApiRequestInfo().getUsername());
+        additionalParams.put(CLIENT_ID_TAG, fsApiRequestContext.getApiRequestInfo().getConsumerKey());
+        additionalParams.put(RESOURCE_PARAMS, getResourceParamMap(fsApiRequestContext));
+
+        JSONObject validationRequest;
+        if (StringUtils.isNotBlank(fsApiRequestContext.getModifiedPayload())) {
+            validationRequest = createValidationRequestPayload(requestHeaders,
+                    fsApiRequestContext.getModifiedPayload(), additionalParams);
+        } else {
+            validationRequest = createValidationRequestPayload(requestHeaders,
+                    fsApiRequestContext.getRequestPayload(), additionalParams);
+        }
+        String enforcementJWTPayload = generateJWT(validationRequest.toString());
+        JSONObject jsonResponse;
+        try {
+            String response = invokeConsentValidationService(enforcementJWTPayload);
+            jsonResponse = new JSONObject(response);
+        } catch (IOException | FinancialServicesException e) {
+            handleError(fsApiRequestContext, FinancialServicesErrorCodes.CONSENT_VALIDATION_REQUEST_FAILURE,
+                    e.getMessage(), FinancialServicesErrorCodes.SERVER_ERROR_CODE);
+            return;
+        }
+
+        boolean isValid = (boolean) jsonResponse.get(IS_VALID);
```

Comment:
getBoolean() ?

Reply:
Fixed in https://github.com/wso2/financial-services-accelerator/pull/130/commits/1e13ec5af065388311c13d8ca7f04ef804d84184